Standard Ebooks.js

```diff
@@ -0,0 +1,227 @@
+{
+	"translatorID": "cc27d642-eab0-4c85-aba8-7a01a9b4e779",
+	"label": "Standard Ebooks",
+	"creator": "Brendan O'Connell",
+	"target": "^https?://standardebooks\\.org/",
+	"minVersion": "5.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2023-02-23 08:52:15"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2023 Brendan O'Connell
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, _url) {
+	// if a book is found, return book
+	if (doc.querySelectorAll('meta[content="book"]').length) {
+		return 'book';
+	}
+	// run getSearchResults function. If it finds an array of rows, will return true
+	// and this else if will return multiple.
+	else if (getSearchResults(doc, true)) {
+		return 'multiple';
+	}
+	return false;
+}
+
+// for multiple results
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	// create an array
+	var rows = doc.querySelectorAll('li[typeof="schema:Book"]')
+	// loop through the array, just scraping URL and title. title is displayed to the user in
+	// Zotero.selectItems popup, URL is sent to scrape function
+	for (let row of rows) {
+		let href = "https://www.standardebooks.org" + row.attributes.about.nodeValue
+		// innerText contains title and author, separated by \n. Delete the author and just display
+		// title to user.
+		let title = row.innerText.slice(0, row.innerText.lastIndexOf('\n'));
```

Comment:
@AbeJellinek 

It looks like the title is contained in a span within an a:

`<a href="[/ebooks/j-s-fletcher/the-talleyrand-maxim](view-source:https://standardebooks.org/ebooks/j-s-fletcher/the-talleyrand-maxim)" property="schema:url"><span property="schema:name">The Talleyrand Maxim</span></a>`

so we need schema:name, rather than schema:url - as currently written, Scaffold is giving me an error, "No title specified for item", which makes sense. How do you select this using the text() helper function?

Reply:
Great, thank you!